lib/flatpak-installation.c

```diff
@@ -1897,7 +1897,8 @@ flatpak_installation_update_full (FlatpakInstallation    *self,
   if (result == NULL)
     goto out;
 
-  if ((flags & FLATPAK_UPDATE_FLAGS_NO_DEPLOY) != 0)
+  if ((flags & FLATPAK_UPDATE_FLAGS_NO_DEPLOY) != 0 &&
+      (flags & FLATPAK_UPDATE_FLAGS_NO_PRUNE) == 0)
```

Comment:
Nitpick: I’d say `!(flags & FLATPAK_UPDATE_FLAGS_NO_PRUNE)` is a bit more obvious to read, since the sense of the condition is not left to the end of the statement. However, it depends on the prevailing flatpak coding style.